src/components/App.jsx

```diff
@@ -0,0 +1,43 @@
+import React, { useEffect } from 'react';
+
+import { useDispatch, useSelector } from 'react-redux';
+
+import CategoriesContainer from '../containers/CategoriesContainer';
+
+import LocationsContainer from '../containers/LocationsContainer';
+
+import RestaurantsContainer from '../containers/RestaurantsContainer';
+
+import {
+  loadLocations,
+  loadCategories,
+  loadRestaurants,
+} from '../actions';
+
+export default function App() {
+  const dispatch = useDispatch();
+
+  const { locationName, categoryName } = useSelector((state) => ({
+    locationName: state.selectedLocationName,
+    categoryName: state.selectedCategoryName,
+  }));
+
+  useEffect(() => {
+    dispatch(loadLocations());
+    dispatch(loadCategories());
+  }, []);
+
+  if (locationName && categoryName) {
+    useEffect(() => {
+      dispatch(loadRestaurants(locationName, categoryName));
+    }, [locationName, categoryName]);
+  }
```

Comment:
locationName && categoryName 인 경우에 loadRestaurants을 dispatch 해주셨네요! 예외처리는 useEffect 내부에서 처리해주는게 좋습니다.

```suggestion
    useEffect(() => {
      if (locationName && categoryName) {
        dispatch(loadRestaurants(locationName, categoryName));
      }
    }, [locationName, categoryName]);
```

그리고 결국 지역과 카테고리를 선택한 경우에 레스토랑 정보를 불러오면 될 것 같아요. `loadRestaurants`을 `dispatch` 해주는 위치가 이곳이 좋을까요?